### PR TITLE
feat(Message) - support Markdown in messages

### LINKED ISSUE #9949: Markdown support - Frontend
* Overview ticket: https://github.com/nextcloud/spreed/issues/1027
* API discussion: https://github.com/nextcloud/spreed/issues/9950

We plan to role out the following support for markdown:

# non-optional
- [ ] bold
- [ ] italic

# optional
Optional means it is just shown with the markdown syntax

- [ ] strike-through
- [ ] list
- [ ] todo list with checkboxes
- [ ] code block using 3 `
- [ ] inline code using `
- [ ] quote
- [ ] headlines
- [ ] dividers `---`

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Close #9949 

### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
image | image
image | image
image | image
image | image
image | image
image | image

### 🚧 Tasks

- [ ] Styles for markdown elements should be defiined
- [ ] Visual check
- [ ] Code review

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not required
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

**Review by nickvergessen — APPROVED** [review]

**Comment by Antreesy** [comment]
- https://github.com/nextcloud/nextcloud-vue/pull/4365 and deps bump is required for support of:
  - autolink literals
  - footnotes
  - strikethrough
  - tables
  - tasklists
=> goes to NC 28

**Review by DorraJaouad — APPROVED** [review]
Visually tested and it works.

File: src/components/MessagesList/MessagesGroup/Message/Message.vue [hunk 1]
```diff
@@ -57,32 +57,34 @@ the main body of the message as well as a quote.
...
 						:reference-limit="0" />
 				</div>
-				<div v-else class="message-body__main__text" :class="{'system-message': isSystemMessage}">
+				<div v-else class="message-body__main__text message-body__main__text--markdown">
 					<Quote v-if="parent" :parent-id="parent" v-bind="quote" />
 					<NcRichText :text="message"
 						:arguments="richParameters"
-						:autolink="true"
+						autolink
+						use-markdown
```
Comment (nickvergessen) [inline src/components/MessagesList/MessagesGroup/Message/Message.vue:87]: Should only be done when the markdown property is passed.
(Also reference limit shoulkd be 0 in that case)

File: src/components/MessagesList/MessagesGroup/Message/Message.vue [hunk 2]
```diff
@@ -57,32 +57,34 @@ the main body of the message as well as a quote.
...
 						:reference-limit="0" />
 				</div>
-				<div v-else class="message-body__main__text" :class="{'system-message': isSystemMessage}">
+				<div v-else class="message-body__main__text message-body__main__text--markdown">
 					<Quote v-if="parent" :parent-id="parent" v-bind="quote" />
 					<NcRichText :text="message"
 						:arguments="richParameters"
-						:autolink="true"
+						autolink
+						:use-markdown="markdown"
```
[the remark this answers is not in the document]
Comment (Antreesy) [reply src/components/MessagesList/MessagesGroup/Message/Message.vue:87]: Also see for prop `markdown`

**Pull request merged**